From c411ca8fff48c17058a24bbfb3329cf5cbc0d52c Mon Sep 17 00:00:00 2001
From: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
Date: Wed, 8 Mar 2023 15:46:04 +0530
Subject: [PATCH] Changes to be in sync with features enabled in linux variant

- Added AV1 encode files
- Enable mfx features enabled in linux
- Updated version informations

Change-Id: I93d14907bfa5658230ebb42846dc0651fba31aa4
Tracked-On: OAM-106726
Signed-off-by: Jeevaka Prabu Badrappan <jeevaka.badrappan@intel.com>
---
 _studio/mfx_lib/Android.mk         | 36 ++++++++++++++++++++++++++++++
 android/include/mfx_android_defs.h | 23 +++++++++----------
 android/mfx_defs.mk                |  7 +++---
 3 files changed, 49 insertions(+), 17 deletions(-)

diff --git a/_studio/mfx_lib/Android.mk b/_studio/mfx_lib/Android.mk
index 913687d6..af9ef0cc 100644
--- a/_studio/mfx_lib/Android.mk
+++ b/_studio/mfx_lib/Android.mk
@@ -40,6 +40,32 @@ MFX_LOCAL_SRC_FILES_HW += $(addprefix mfx_lib/ext/genx/h264_encode/isa/, \
     genx_simple_me_gen12lp_isa.cpp \
     genx_histogram_gen12lp_isa.cpp)
 
+MFX_LOCAL_SRC_FILES_HW += \
+    mfx_lib/encode_hw/av1/av1ehw_disp.cpp \
+    mfx_lib/encode_hw/av1/agnostic/av1ehw_base.cpp \
+    mfx_lib/encode_hw/av1/linux/base/av1ehw_base_lin.cpp \
+    mfx_lib/encode_hw/av1/linux/base/av1ehw_base_va_lin.cpp \
+    mfx_lib/encode_hw/av1/linux/base/av1ehw_base_va_packer_lin.cpp \
+    mfx_lib/encode_hw/av1/linux/base/av1ehw_base_max_frame_size_lin.cpp \
+    mfx_lib/encode_hw/av1/agnostic/Xe_HPM/av1ehw_xe_hpm_segmentation.cpp \
+    mfx_lib/encode_hw/av1/linux/Xe_HPM/av1ehw_xe_hpm_lin.cpp \
+    mfx_lib/encode_hw/av1/linux/Xe_LPM_plus/av1ehw_xe_lpm_plus_lin.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_alloc.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_constraints.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_qmatrix.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_general.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_general_defaults.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_impl.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_packer.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_query_impl_desc.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_segmentation.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_task.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_tile.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_encoded_frame_info.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_max_frame_size.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_enctools.cpp \
+    mfx_lib/encode_hw/av1/agnostic/base/av1ehw_base_hdr.cpp \
+
 MFX_LOCAL_SRC_FILES_HW += \
     mfx_lib/encode_hw/hevc/hevcehw_disp.cpp \
     mfx_lib/encode_hw/hevc/agnostic/hevcehw_base.cpp \
@@ -48,6 +74,7 @@ MFX_LOCAL_SRC_FILES_HW += \
     mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_constraints.cpp \
     mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_dirty_rect.cpp \
     mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_encoded_frame_info.cpp \
+    mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_encode_stats.cpp \
     mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_enctools.cpp \
     mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_ext_brc.cpp \
     mfx_lib/encode_hw/hevc/agnostic/base/hevcehw_base_hdr_sei.cpp \
@@ -118,6 +145,15 @@ MFX_LOCAL_INCLUDES_HW := \
     $(MFX_HOME)/_studio/mfx_lib/ext/genx/asc/isa \
     $(MFX_HOME)/_studio/mfx_lib/ext/h264/include \
     $(MFX_HOME)/_studio/mfx_lib/ext/mpeg2/include \
+    $(MFX_HOME)/_studio/mfx_lib/encode_hw/av1 \
+    $(MFX_HOME)/_studio/mfx_lib/encode_hw/av1/agnostic \
+    $(MFX_HOME)/_studio/mfx_lib/encode_hw/av1/agnostic/base \
+    $(MFX_HOME)/_studio/mfx_lib/encode_hw/av1/agnostic/Xe_HPM \
+    $(MFX_HOME)/_studio/mfx_lib/encode_hw/av1/agnostic/Xe_LPM_plus \
+    $(MFX_HOME)/_studio/mfx_lib/encode_hw/av1/linux \
+    $(MFX_HOME)/_studio/mfx_lib/encode_hw/av1/linux/base \
+    $(MFX_HOME)/_studio/mfx_lib/encode_hw/av1/linux/Xe_HPM \
+    $(MFX_HOME)/_studio/mfx_lib/encode_hw/av1/linux/Xe_LPM_plus \
     $(MFX_HOME)/_studio/mfx_lib/encode_hw/hevc \
     $(MFX_HOME)/_studio/mfx_lib/encode_hw/hevc/agnostic \
     $(MFX_HOME)/_studio/mfx_lib/encode_hw/hevc/agnostic/base \
diff --git a/android/include/mfx_android_defs.h b/android/include/mfx_android_defs.h
index 026268e6..7052485a 100644
--- a/android/include/mfx_android_defs.h
+++ b/android/include/mfx_android_defs.h
@@ -21,40 +21,37 @@
 #ifndef __MFX_ANDROID_DEFS_H__
 #define __MFX_ANDROID_DEFS_H__
 
-  #define MFX_ENABLE_VC1_VIDEO_DECODE
+  #define MFX_ENABLE_ASC
+  #define MFX_ENABLE_CPLIB
 
-  #define MFX_ENABLE_MPEG2_VIDEO_DECODE
+  #define MFX_ENABLE_VPP
 
-  #define MFX_ENABLE_H265_VIDEO_DECODE
+  #define MFX_ENABLE_VC1_VIDEO_DECODE
 
+  #define MFX_ENABLE_H265_VIDEO_DECODE
   #define MFX_ENABLE_H265_VIDEO_ENCODE
 
   #define MFX_ENABLE_H264_VIDEO_DECODE
-
   #define MFX_ENABLE_H264_VIDEO_ENCODE
 
   #define MFX_ENABLE_MJPEG_VIDEO_DECODE
-
   #define MFX_ENABLE_MJPEG_VIDEO_ENCODE
 
+  #define MFX_ENABLE_MPEG2_VIDEO_ENCODE
+  #define MFX_ENABLE_MPEG2_VIDEO_DECODE
+
   #define MFX_ENABLE_VP8_VIDEO_DECODE
 
   #define MFX_ENABLE_VP9_VIDEO_DECODE
-
   #define MFX_ENABLE_VP9_VIDEO_ENCODE
 
   #define MFX_ENABLE_AV1_VIDEO_DECODE
-
-  #define MFX_ENABLE_VPP
+  #define MFX_ENABLE_AV1_VIDEO_ENCODE
 
   #define MFX_ENABLE_EXT
 
-  //#define MFX_ENABLE_ENCTOOLS
-
-  //#define MFX_ENABLE_USER_ENCTOOLS
-
 #if MFX_ANDROID_VERSION >= MFX_P
   #define MFX_ENABLE_KERNELS
 #endif
 
-#endif // #ifndef __MFX_ANDROID_DEFS_H__
\ No newline at end of file
+#endif // #ifndef __MFX_ANDROID_DEFS_H__
diff --git a/android/mfx_defs.mk b/android/mfx_defs.mk
index 2397ac3d..5707e4a4 100644
--- a/android/mfx_defs.mk
+++ b/android/mfx_defs.mk
@@ -14,11 +14,11 @@
 MFX_CFLAGS := -DANDROID
 
 #Media Version
-MEDIA_VERSION := 20.5
+MEDIA_VERSION := 23.1.0
 MEDIA_VERSION_EXTRA := ""
 MEDIA_VERSION_ALL := $(MEDIA_VERSION).pre$(MEDIA_VERSION_EXTRA)
 
-MFX_CFLAGS += -DMEDIA_VERSION_STR=\"\\\"${MEDIA_VERSION_ALL}\\\"\"
+MFX_CFLAGS += -DMEDIA_VERSION_STR=\"\\\"${MEDIA_VERSION}\\\"\"
 MFX_CFLAGS += -DONEVPL_EXPERIMENTAL
 
 # Android version preference:
@@ -48,8 +48,7 @@ MFX_CFLAGS += \
   -include mfx_android_config.h
 
 MFX_CFLAGS += \
-    -DMFX_ENABLE_CPLIB \
-    -DMFX_VERSION=2006
+    -DMFX_VERSION=2008
 
 MFX_CFLAGS += \
   -DMFX_FILE_VERSION=\"`echo $(MFX_VERSION) | cut -f 1 -d.``date +.%-y.%-m.%-d`\" \
-- 
2.17.1

